pdns/dnsdist-idstate.hh

```diff
@@ -241,7 +241,7 @@ struct IDState
   std::unique_ptr<QTag> qTag{nullptr}; // 8
   boost::optional<uint32_t> tempFailureTTL; // 8
   const ClientState* cs{nullptr}; // 8
-  DOHUnit* du{nullptr}; // 8
+  DOHUnit* du{nullptr}; // 8 (not a unique_ptr because we currently need to be able to peek at it without knowing taking ownership until later)
```

Comment:
Suggestion: use `duNaked` or something like that to make it very clear it's not a smart pointer when using it as other instances of `du` hold a `unique_pointer`.